pkg/snapstore/swift_snapstore.go

```diff
@@ -146,35 +176,29 @@ func uploadChunk(s *SwiftSnapStore, snap *Snapshot, file *os.File, offset, chunk
 	return res.Err
 }
 
-func retryChunkUpload(s *SwiftSnapStore, snap *Snapshot, file *os.File, offset, chunkSize int64, errCh chan<- chunkUploadError) {
-	var (
-		maxAttempts uint = 5
-		curAttempt  uint = 1
-		err         error
-	)
+func (s *SwiftSnapStore) chunkUploader(wg *sync.WaitGroup, stopCh <-chan struct{}, snap *Snapshot, file *os.File, chunkUploadCh chan chunk, errCh chan<- chunkUploadResult) {
+	defer wg.Done()
 	for {
-		logrus.Infof("Uploading chunk with offset : %d, attempt: %d", offset, curAttempt)
-		err = uploadChunk(s, snap, file, offset, chunkSize)
-		logrus.Infof("For chunk upload of offset %d, err %v", offset, err)
-		if err == nil || curAttempt == maxAttempts {
-			break
+		select {
+		case <-stopCh:
+			return
+		case chunk, more := <-chunkUploadCh:
+			if !more {
+				return
+			}
+			logrus.Infof("Uploading chunk with offset : %d, attempt: %d", chunk.offset, chunk.attempt)
+			err := s.uploadChunk(snap, file, chunk.offset, chunk.size)
+			logrus.Infof("For chunk upload of offset %d, err %v", chunk.offset, err)
```

Comment:
Check comment above.

Reply:
I have addressed it already. Please refresh your PR and check.